redis/src/cluster_async/mod.rs

```diff
@@ -1094,40 +1124,40 @@ where
                     pipeline,
                     offset,
                     count,
-                    Self::get_connection(info.redirect, route, core, asking),
+                    Self::get_connection(route, core),
                 )
                 .await
             }
         }
     }
 
     async fn get_connection(
-        mut redirect: Option<Redirect>,
-        route: SingleNodeRoutingInfo,
+        routing: SingleNodeRouting<C>,
         core: Core<C>,
-        asking: bool,
     ) -> (ConnectionIdentifier, C) {
         let read_guard = core.conn_lock.read().await;
+        let asking = matches!(routing, SingleNodeRouting::Redirect(Redirect::Ask(_)));
```

Comment:
Instead of doing this check on each call for get_connection, can you set it with false and only set it to true in the relevant SingleNodeRouting::Redirect(Redirect::Ask(ask_addr)) match?

Reply:
done, but IMO it's not even an optimization - the compiler should notice that both branches match, and combine the logic itself.